packages/blocks/src/store/selectors.js

```diff
@@ -40,15 +51,8 @@ const getNormalizedBlockType = ( state, nameOrType ) =>
  * @return {Array} Block Types.
  */
 export const getBlockTypes = createSelector(
-	( state ) => {
-		return Object.values( state.blockTypes ).map( ( blockType ) => {
-			return {
-				...blockType,
-				variations: getBlockVariations( state, blockType.name ),
-			};
-		} );
-	},
-	( state ) => [ state.blockTypes, state.blockVariations ]
+	( state ) => Object.values( state.blockTypes ),
```

Comment:
I noticed that we don't combine all registered block variations with those registered with the block settings in `getBlockType`:

https://github.com/WordPress/gutenberg/blob/f242c91b21bbbccf0a0129e4946aadfa4d04baca/packages/blocks/src/store/selectors.js#L62-L64

For consistency, we should use one approach.